src/analysis/utils/tool-utils.ts

```diff
@@ -222,7 +228,11 @@ export const allAppTypes: AppToolLabel[] = _.flow(_.map('label'), _.compact)(app
 export const isPauseSupported = (toolLabel: ToolLabel): boolean =>
   !_.find((tool: AppTool | RuntimeTool) => tool.label === toolLabel)(tools)?.isPauseUnsupported;
 
-export const isToolHidden = (toolLabel: ToolLabel, cloudProvider: CloudProvider): boolean =>
+export const isToolHidden = (
+  toolLabel: ToolLabel,
+  cloudProvider: CloudProvider,
+  currentApp: App | undefined
```

Comment:
Hmm... it doesn't feel right to add the current app to the method signature. In my head this method should be able to answer this question with cloud platform + tool + feature preview enablement state, regardless of the current app. If the currentApp state (in this case the current app not existing) causes something to be hidden the visual component should be responsible for knowing that as I see it. Thoughts?

Reply:
I do not like this generic utility method needing to keep track of additional state. I would really like to de-nest this particular part of the conditional https://github.com/DataBiosphere/terra-ui/pull/4339/files#diff-c014b6d2c74c44670a941ef25e49559db7d3d8fb7552362ca78fc801a6ed02bdR255

Reply:
Would you feel better if the app-specific logic was pulled into an `isAppHidden` helper that took in a list of `apps`, and then delegated to `isToolHidden` in the default case (without knowledge of apps)? Otherwise we'll just be breaking out the conditional check into most of the different usages, making it harder to keep consistent in the future. This would also allow us to condense some of the `getCurrentAppForUser` vs `getCurrentApp` logic that currently exists prior to passing into `isToolHidden`.